bound/doc.go

```diff
@@ -0,0 +1,6 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package bound provides types for bounded data.
```

Comment:
Are there any other things that this package provides? Do we possibly want ℝ² and ℝ³ bounds in the future to go along with ℝ¹?

Reply:
Possibly... `distmv` and `optimize` need arbitrary dimensional `[]bound.Bound` though. We don't have many explicitly dimensioned things in Gonum at the moment. This is a natural type to go in `gonum.org/v1/gonum`, but we've resisted that for a variety of reasons, and we don't want a `util` package, so I wasn't sure where else it could go. I was thinking in the future it could possibly have some intersection/union style helper routines, sort of like in the `distmv.Uniform.Entropy` calculation.

Reply:
Yeah, I think intersection, union, Jacquard are probably enough reason for having this as a seprate package.